src/Mvc/Mvc.Core/src/ControllerBase.cs

```diff
@@ -1924,6 +1924,7 @@ public virtual ConflictObjectResult Conflict([ActionResultObjectValue] ModelStat
     /// </summary>
     /// <returns>The created <see cref="BadRequestObjectResult"/> for the response.</returns>
     [NonAction]
+    [DefaultStatusCode(StatusCodes.Status400BadRequest)]
```

Comment:
I don't mind this, but is it super tricky to put the attribute on the return value (because it's kinda more appropriate). Either way it's super nit-picky.

Reply:
So, that's what I was originally trying to do but I ran into limitations with the Roslyn API when parsing the return method. If you have an action like this:

```csharp
public IActionResult Foo()
{
  return ValidationProblem();
}
```

We're able to identify that `ValidationProblem` was invoked and examine the attributes on the `ValidationProblem` method pretty easily.

If the `ValidationProblem` looked something like this:

```csharp
public IActionResult ValidationProblem()
{
   return new FooResult(); // Where `FooResult` has `DefaultStatusCode` attribute
}
```

We can't peek into the type returned from `ValidationProblem` and examine the attributes on it, particularly for the case of `ValidationProblem` since that method is defined in an assembly separate from the source assembly (and probably other shorthand methods as well).